physionet-django/project/modelcomponents/metadata.py

```diff
@@ -581,12 +581,13 @@ class Reference(models.Model):
     content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
     object_id = models.PositiveIntegerField()
     project = GenericForeignKey('content_type', 'object_id')
+    order = models.PositiveIntegerField(null=True)
 
     description = models.CharField(max_length=1000)
 
     class Meta:
         default_permissions = ()
-        unique_together = (('description', 'content_type', 'object_id'),)
+        unique_together = (('description', 'content_type', 'order'),)
```

Comment:
@tompollard  I am not sure if the unique is currently setup in the best possible way. Our current unique setup will cause false duplicate error in case where the reference and order of reference is same in two different projects(of same project type).

For example

Project A ( contentType=database)

| Order | reference | uniqueKey      |
|-------|-----------|----------------|
| 1     | 'a'       | (27, 'a', 1)   |
| 2     | 'b'       | (27, 'b', 2)   |
| 3     | 'c'       | (27, 'c', 3)   |
| 4     | 'd'       | (27, 'd', 4)   |


Project B ( contentType=database)

| Order | reference | uniqueKey      |
|-------|-----------|----------------|
| 1     | 'aa'       | (27, 'aa', 1)  |
| 2     | 'b'       | (27, 'b', 2)   |
| 3     | 'cc'      | (27, 'cc', 3)  |
| 4     | 'dd'      | (27, 'dd', 4)  |


as you can see we will have a conflict on the 2nd reference.


I think project might be better instead of content_type.

Reply:
maybe not `project` exactly, as the generic foreign key throws error when trying to use in `unique_together`

Reply:
not sure if adding object_id is a good idea as that would most likely allow is to insert duplicates

for example
| Order | reference | uniqueKey                | content_id |
|-------|-----------|--------------------------|------------|
| 1     | 'a'       | (27, 'a', 1,11)    | 11   |
| 2     | 'b'       | (27, 'b', 2,12)    | 12   |
| 3     | 'c'       | (27, 'c', 3,13)    | 13   |
| 4     | 'd'       | (27, 'd', 4,14)    | 14   |
| 5     | 'a'       | (27, 'a', 5,15)    | 15   |


Although the front end will not send the duplicate reference, but updating the unique_together this way would make zero sense, as it would allow duplicate in database

Reply:
> Our current unique setup will cause false duplicate error in case where the reference and order of reference is same in two different projects(of same project type).

Yes, good catch, you're right! This doesn't make sense. 

> I think project might be better instead of content_type ... maybe not project exactly, as the generic foreign key throws error when trying to use in unique_together

I agree, `content_type` doesn't add anything useful here. I also agree that project would be a good alternative (and it's a shame that generic foreign key's make this difficult.

> not sure if adding object_id to unique_together is a good idea as that would most likely allow is to insert duplicates

Unless I'm missing something, I think it is fine to switch `object_id` with `content_type`. It is true that we may end up with duplicate reference strings in the database (slightly inefficient in terms of storage) but I don't think this is too much to worry about.

In summary, I think you should be good to change `content_type` to `object_id`.

Reply:
Alright, i think that makes sense `object_id` instead of `content_type` should work. The object ID is a integer that increments automatically. So it should help us prevent duplicate across different project.(and the order should help fix the reference error that we were trying to fix)

I will do more tests, and submit a PR in the monring

Thanks @tompollard